adapters/invibes/invibes.go

```diff
@@ -0,0 +1,327 @@
+package invibes
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/golang/glog"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbsmetrics"
+)
+
+const adapterVersion = "prebid_1.0.0"
+
+type InvibesAdRequest struct {
+	BidParamsJson string
+	Location      string
+	Lid           string
+	IsTestBid     bool
+	Kw            string
+	IsAmp         bool
+	Width         string
+	Height        string
+	GdprConsent   string
+	Gdpr          bool
+	Bvid          string
+	InvibBVLog    bool
+	VideoAdDebug  bool
+}
+type InvibesBidParams struct {
+	PlacementIds []string
+	BidVersion   string
+	Properties   map[string]InvibesPlacementProperty
+}
+type InvibesPlacementProperty struct {
+	Formats []openrtb.Format
+	ImpId   string
+}
+type InvibesInternalParams struct {
+	BidParams   InvibesBidParams
+	Host        string
+	IsAmp       bool
+	Gdpr        string
+	GdprConsent string
+
+	TestIp   string
+	TestBvid string
+	TestLog  bool
+}
+type BidServerBidderResponse struct {
+	Currency  string              `json:"currency"`
+	TypedBids []BidServerTypedBid `json:"typedBids"`
+	Error     string              `json:"error"`
+}
+type BidServerTypedBid struct {
+	Bid          openrtb.Bid `json:"bid"`
+	DealPriority int         `json:"dealPriority"`
+}
+
+func (a *InvibesInternalParams) IsTestRequest() bool {
+	return a.TestIp != "" || a.TestBvid != ""
+}
+
+type InvibesAdapter struct {
+	EndpointTemplate template.Template
+}
+
+func NewInvibesBidder(endpointTemplate string) *InvibesAdapter {
+	urlTemplate, err := template.New("endpointTemplate").Parse(endpointTemplate)
+	if err != nil {
+		glog.Fatal("Unable to parse endpoint url template")
+		return nil
+	}
+	return &InvibesAdapter{EndpointTemplate: *urlTemplate}
+}
+
+func (a *InvibesAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No imps in the bid request",
+		}}
+	}
+
+	consentString := ""
+	if request.User != nil {
+		var extUser openrtb_ext.ExtUser
+		if err := json.Unmarshal(request.User.Ext, &extUser); err == nil {
+			consentString = extUser.Consent
+		}
+	}
+	gdprApplies := "1"
```

Comment:
Does Invibes only serve EU traffic? Defaulting gdprApplies to "1" and the consentString to "" will invalidate a lot of non-EU requests.

Reply:
The documentation for GDPR property is: 
GDPR should be "1" if the caller believes the user is subject to GDPR laws, "0" if not, and undefined if it's unknown. 
Can we consider that in case of EU requests the GDPR property is always defined?

Reply:
The "core" of PBS does the best it can to infer this property. If the request doesn't explicitly set it, it'll look at the request's geo information (if available), and otherwise will fall back on host-level geo settings (a lot less precise). I believe it's possible for EU requests to still have an unknown GDPR applies at this point though.

Does that sound about right @hhhjort ?

Reply:
Thank you for the clarification.